Add compensation integration (#15192)
home-assistant/home-assistant.github.io: source/_integrations/compensation.markdown

---
title: Compensation
description: Instructions on how to integrate compensation sensors into Home Assistant.
ha_category:
  - Utility
  - Sensor
ha_iot_class: Calculated

> ha_release: "2021.4.0"

Reviewer: ```suggestion
ha_release: "2021.5"
```

---
title: Compensation
description: Instructions on how to integrate compensation sensors into Home Assistant.
ha_category:
  - Utility
  - Sensor
ha_iot_class: Calculated

> ha_release: '0.117'

Reviewer: ```suggestion
ha_release: "2021.4.0"
```

---
title: Compensation
description: Instructions on how to integrate compensation sensors into Home Assistant.
ha_category:
  - Utility
  - Sensor

> ha_iot_class: Local Push

Reviewer: This integration calculates its states based on existing data.

```suggestion
ha_iot_class: Calculated
```

Author: Yep, I assumed that was 'local push'.  Makes more sense as calculated.

ha_release: '0.117'
ha_quality_scale: internal
ha_codeowners:
  - '@petro31'
ha_domain: compensation
---

> The `compensation` sensor platform consumes the state from other sensors. It exports the `compensated` value as state and the following values as attributes: `entity_id` and `coefficients`.

Reviewer: ```suggestion
The Compensation integration consumes the state from other sensors. It exports the compensated value as state and the following values as attributes: `entity_id` and `coefficients`.
```

## Configuration

To enable the compensation sensor, add the following lines to your `configuration.yaml`:

```yaml
# Example configuration.yaml entry
sensor:

>   - platform: compensation

Reviewer: We no longer accept YAML configuration for integrations on a platform level.

For more information about this decision see:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>

Author: Ok, I'll probably need to update the other PR to handle that.

Author: fixed

    name: Yamaha Zone 1 Volume
    entity_id: media_player.yamaha_receiver
    attribute: volume_level

>     unit_of_measurement: dB

Reviewer: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

Author: fixed

    data_points:
      - 0.2 -> -80.0
      - 1.0 -> 0.0
```

{% configuration %}
entity_id:
  description: The entity to monitor.
  required: true
  type: string
data_points:

>   description: The collection of data point conversions with the format `uncompensated_value -> compensated_value`.  e.g. `1.0 -> 2.1`. The number of required data points is equal to the polynomial `degree` + 1. For example, a linear compensation (with `degree: 1`) requires at least 2 data points.

Reviewer 2: ```suggestion
  description: "The collection of data point conversions with the format `uncompensated_value -> compensated_value`.  e.g. `1.0 -> 2.1`. The number of required data points is equal to the polynomial `degree` + 1. For example, a linear compensation (with `degree: 1`) requires at least 2 data points."
```

>   description: "The collection of data point conversions with the format `uncompensated_value -> compensated_value`.  e.g. `1.0 -> 2.1`. The number of required data points is equal to the polynomial `degree` + 1. For example, a linear compensation (with `degree: 1`) requires at least 2 data points."

Reviewer: ```suggestion
  description: "The collection of data point conversions with the format `uncompensated_value -> compensated_value`.  e.g., `1.0 -> 2.1`. The number of required data points is equal to the polynomial `degree` + 1. For example, a linear compensation (with `degree: 1`) requires at least 2 data points."
```

  required: true
  type: list
name:
  description: Name of the sensor to use in the frontend.
  required: false
  default: Compensation
  type: string
attribute:
  description: Attribute to monitor.
  required: false
  type: string
degree:

>   description: The degree of a polynomial. e.g. Linear compensation (y = x + 3) has 1 degree, Quadratic compensation (y = x<sup>2</sup> + x + 3) has 2 degrees, etc.

Reviewer: ```suggestion
  description: "The degree of a polynomial. e.g., Linear compensation (`y = x + 3`) has 1 degree, Quadratic compensation (`y = x<sup>2</sup> + x + 3`) has 2 degrees, etc."
```

Author: will the `` kill the x<sup>2</sup>?  That's why I avoided it in the first place.

Author: `test<sup>test</sup>` vs test<sup>test</sup>

Author: Looks like it does

Author: I left the equations as non-code to ensure the superscript stays.
